agent/bench-scripts/pbench-fio

```diff
@@ -601,7 +601,7 @@ function fio_run_job() {
 			if pbench-is-local ${client}; then
 				debug_log "killing any old local fio processes and starting new a fio process for the local client"
 				killall ${benchmark} > /dev/null 2>&1
-				screen -dmS fio-server bash -c '${benchmark_bin} --server=,${client_ports[${client}]} > client-result.txt 2> client-result.err'
+				screen -dmS fio-server bash -c "${benchmark_bin} --server=,${client_ports[${client}]} > '${benchmark_results_dir}/client-result.txt' 2> '${benchmark_results_dir}/client-result.err'"
```

Comment:
I am not sure that is what we want.  Take a look at the ssh one.  I think we want to have:

```
screen -dmS fio-server bash -c "'${benchmark_bin} --server=,${client_ports[${client}]} > ${benchmark_results_dir}/client-result.txt 2> ${benchmark_results_dir}/client-result.err'"
```

Basically, we want the entire command in the `bash -c` execution.

Reply:
I don't see a functional difference between Peter's and Lukas's approaches, other than that Peter's protects against embedded spaces (et al.) in `${benchmark_bin}` -- other than that, I _think_ they produce equivalent results.

However, my inclination is to have the local and remote command lines match as much as possible.  So, for starters, I recommend inserting the `cd ${benchmark_results_dir}` into the pipeline with `screen`, rather than using absolute paths in the `screen` command.  But, beyond that, if we can figure out how to get the quoting right, I'd like to see the command line built once for both cases, and then just used either directly or via `ssh` for the two cases.  (I mean, I think we want to do the same thing in both cases, so why not use the same command line?...if we can....)

But, perhaps that's too much to ask.  🥴

Reply:
I can follow-up with another PR to try to address that.

Reply:
Let's see if PR https://github.com/ldoktor/pbench/pull/8 is useful.

Reply:
> I am not sure that is what we want. Take a look at the ssh one. I think we want to have:
> 
> ```
> screen -dmS fio-server bash -c "'${benchmark_bin} --server=,${client_ports[${client}]} > ${benchmark_results_dir}/client-result.txt 2> ${benchmark_results_dir}/client-result.err'"
> ```
> 
> Basically, we want the entire command in the `bash -c` execution.

This command won't work because of double-quotation. Let's simplify it a bit to `screen -dmS test bash -c "'echo a a a > /tmp/with spaces'"` which results in executing `["bash", "-c", "'echo a a a > /tmp/with spaces'"]`.

Double quotes only won't work either, because of the spaces: `screen -dmS test bash -c "echo a a a > /tmp/with spaces"`.

Safer solution is to escape each element: `screen -dmS test bash -c "echo 'a a a' > '/tmp/with spaces'"`. But even I was not that paranoiac to escape the ports and benchmark bin (although I probably should have).